ql/src/test/queries/clientpositive/prune_empty_table.q

```diff
@@ -0,0 +1,24 @@
+set hive.tez.cartesian-product.enabled=true;
+
+create table c (a1 int) stored as orc;
+create table tmp1 (a int) stored as orc;
+create table tmp2 (a int) stored as orc;
+
+insert into table c values (3);
+insert into table tmp1 values (3);
+
+explain cbo
+with
+first as (
+select a1 from c where a1 = 3
+),
```

Comment:
CTEs make the query more complex and add another dimension to the problem. If possible let's just skip it.

Can't we repro the problem with just two tables that one of them is empty? Do we need union all + 3 tables?

Since the query was failing at runtime consider adding also a full execution of the query with the results.

Reply:
I will simplify the test and also add another query without explain.